lib/filewatch/discover.rb

```diff
@@ -0,0 +1,141 @@
+require_relative 'watched_file'
+
+module FileWatch
+  class Discover
+    attr_reader  :logger, :files, :wf_vars
+
+    def initialize(opts, loggr)
+      @logger = loggr
+      @watching = []
+      @exclude = []
+      @files = {}
+      set_ignore_older(opts[:ignore_older])
+      set_close_older(opts[:close_older])
+      @wf_vars = [opts[:delimiter], @ignore_older, @close_older]
+      exclude(opts[:exclude])
+    end
+
+    def logger=(loggr)
+      @logger = loggr
+      @converter.logger = loggr
+    end
+
+    def add_converter(converter)
+      @converter = converter
+    end
+
+    def add_path(path)
+      return if @watching.member?(path)
+      @watching << path
+      discover_file(path) do |fpath, stat|
+        WatchedFile.new_initial(*wf_args(fpath, stat)).init_vars(*wf_vars)
+      end
+    end
+
+    def discover
+      @watching.each do |path|
+        discover_file(path) do |fpath, stat|
+          WatchedFile.new_ongoing(*wf_args(fpath, stat)).init_vars(*wf_vars)
+        end
+      end
+    end
+
+    def delete(paths)
+      paths.each {|f| @files.delete(f)}
+    end
+
+    def close_all
+      @files.values.each(&:file_close)
+    end
```

Comment:
true, but we dont actually use the `value` or pass the `value` to a block so in this call I think it should be a `#map` ?